OpenSim/Tools/InverseKinematicsTool.cpp

```diff
@@ -576,3 +516,71 @@ void InverseKinematicsTool::updateFromXMLNode(SimTK::Xml::Element& aNode, int ve
     }
     Object::updateFromXMLNode(aNode, versionNumber);
 }
+
+void InverseKinematicsTool::populateReferences(MarkersReference& markersReference,
+    SimTK::Array_<CoordinateReference>&coordinateReferences) const
+{
+    FunctionSet *coordFunctions = NULL;
+    // Load the coordinate data
+    // bool haveCoordinateFile = false;
+    if (_coordinateFileName != "" && _coordinateFileName != "Unassigned") {
+        Storage coordinateValues(_coordinateFileName);
+        // Convert degrees to radian (TODO: this needs to have a check that the storage is, in fact, in degrees!)
+        _model->getSimbodyEngine().convertDegreesToRadians(coordinateValues);
+        // haveCoordinateFile = true;
+        coordFunctions = new GCVSplineSet(5, &coordinateValues);
+    }
+
+    Set<MarkerWeight> markerWeights;
+    // Loop through old "IKTaskSet" and assign weights to the coordinate and marker references
+    // For coordinates, create the functions for coordinate reference values
+    int index = 0;
+    for (int i = 0; i < _ikTaskSet.getSize(); i++) {
+        if (!_ikTaskSet[i].getApply()) continue;
+        if (IKCoordinateTask *coordTask = dynamic_cast<IKCoordinateTask *>(&_ikTaskSet[i])) {
+            CoordinateReference *coordRef = NULL;
+            if (coordTask->getValueType() == IKCoordinateTask::FromFile) {
+                if (!coordFunctions)
+                    throw Exception("InverseKinematicsTool: value for coordinate " + coordTask->getName() + " not found.");
+
+                index = coordFunctions->getIndex(coordTask->getName(), index);
+                if (index >= 0) {
```

Comment:
`Set::getIndex()` will begin searching at the provided index, but will still wrap to the beginning if the target isn't in `[index, end]`. It looks like the only chance this method would produce unpredictable results is if there are two objects in the Set with the same name, whereupon failure of this method would be the least of our worries.